### PR TITLE
Merged Pull Request

### PR DESCRIPTION
**Pull request by KWang31**
I have closed a pull request and opened a new one because I don't know how to change more files without starting a new one.

### CODE REVIEW

**Comment by bqi343** [comment]
What happened to the [frontmatter](https://github.com/cpinitiative/usaco-guide/blob/master/docs/Content%20Documentation.md#3-frontmatter)?

**Comment by KWang31** [comment]
> What happened to the [frontmatter](https://github.com/cpinitiative/usaco-guide/blob/master/docs/Content%20Documentation.md#3-frontmatter)?

Fixed.

**Comment by bqi343** [comment]
Doesn't render properly with liveupdate.

**Comment by KWang31** [comment]
> Doesn't render properly with liveupdate.

Fixed, I won't start files from scratch again.

**Comment by bqi343** [comment]
You should explain what A, B, C are. One line is missing a period. Use $\mathcal{O}$ instead of $O$. Use 

```java
java code

public static
``` 

instead of 

```
java code

public static
```

**Review by dolphingarlic — CHANGES_REQUESTED** [review]
Some grammar and style changes

**Comment by KWang31** [comment]
Sorry for making 3 commits. Please only check the last one.

File: solutions/JOI 18-snake-escaping.mdx [hunk 1]
```diff
@@ -0,0 +1,135 @@
+---
+id: joi-18-snake-escaping
+title: Snake Escaping Editorial
+source: JOI 18 Final Round Problem 5
```
[the remark this answers is not in the document]
Comment (bqi343) [reply solutions/JOI 18-snake-escaping.mdx:4]: `JOI 2018 Final Round` for source is also fine

**Comment by bqi343** [comment]
Still haven't resolved all the comments.

File: solutions/JOI 18-snake-escaping.mdx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+id: joi-18-snake-escaping
+title: Snake Escaping
+source: JOI 18 Final Round
+author: Kai Wang
+
+---
+
+We use SOS DP to compute $super[mask]$ and $sub[mask]$, which denote $\sum\limits_{mask\in S} a_S$ and $\sum\limits_{S\in mask} a_S$ respectively.
+
+For a query, let $A$ be the set of $0$'s, $B$ be the set of $1$'s, and $C$ be the set of $?$'s. By Pigeonhole principle, $min\{A,B,C\}\le 6$, and I claim we can answer queries in $O(2^{min\{A,B,C\}})$.
```
Comment (bqi343) [inline solutions/JOI 18-snake-escaping.mdx:11]: Use \min instead of min

File: solutions/JOI 18-snake-escaping.mdx [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+If a query has $|C|\le 6$, we can brute force over all the subsets with 0's at A and 1's at B. Since there are $2^{|C|}$ subsets, this takes $\mathcal{O}(2^{|C|})$ operations.
+
+Otherwise, we can answer queries with the principle of inclusion-exclusion. The answer is simply $\sum\limits_{a\in A} super[B | a] (-1)^{|a|}$, which we can calculate in $\mathcal{O}(2^{|A|})$
```
Comment (bqi343) [inline solutions/JOI 18-snake-escaping.mdx:15]: missing a period

File: solutions/JOI 18-snake-escaping.mdx [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+The answer is also $\sum\limits_{b\in B} sub[C | b] (-1)^{|B|-|b|}$, which we can calculate in $\mathcal{O}(2^{|B|})$. They can be computed efficiently with submask enumeration. For a detailed explanation on how submask enumeration works, check https://cp-algorithms.com/algebra/all-submasks.html 
+
+The overall time complexity is $\mathcal{O}(2^L\cdot L+2^{\frac{L}{3}}Q)$ since we compute $super$ and $sub$ in $\mathcal{O}(2^L\cdot L)$ with SOS DP, and we process each query in $\mathcal{O} (2^{\frac L3})$.
```
Comment (bqi343) [inline solutions/JOI 18-snake-escaping.mdx:19]: The other modules (ex. https://github.com/cpinitiative/usaco-guide/blob/ee78824031eb37302c8eade6e732a56f4e8f63c2/solutions/USACO%20feb19-cowland.mdx) put "Time Complexity"  in bold, do that here as well.

File: solutions/JOI 18-snake-escaping.mdx [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+We use SOS DP to compute $super[mask]$ and $sub[mask]$, which denote $\sum\limits_{mask\in S} a_S$ and $\sum\limits_{S\in mask} a_S$ respectively.
+
+For a query, let $A$ be the set of $0$'s, $B$ be the set of $1$'s, and $C$ be the set of $?$'s. By Pigeonhole principle, $min\{A,B,C\}\le 6$, and I claim we can answer queries in $O(2^{min\{A,B,C\}})$.
+
+If a query has $|C|\le 6$, we can brute force over all the subsets with 0's at A and 1's at B. Since there are $2^{|C|}$ subsets, this takes $\mathcal{O}(2^{|C|})$ operations.
+
+Otherwise, we can answer queries with the principle of inclusion-exclusion. The answer is simply $\sum\limits_{a\in A} super[B | a] (-1)^{|a|}$, which we can calculate in $\mathcal{O}(2^{|A|})$
+
+The answer is also $\sum\limits_{b\in B} sub[C | b] (-1)^{|B|-|b|}$, which we can calculate in $\mathcal{O}(2^{|B|})$. They can be computed efficiently with submask enumeration. For a detailed explanation on how submask enumeration works, check https://cp-algorithms.com/algebra/all-submasks.html
```
Comment (bqi343) [inline solutions/JOI 18-snake-escaping.mdx:17]: Use `[description](link)` instead of just pasting the link.

**Review by bqi343 — CHANGES_REQUESTED** [review]

**Pull request merged**